templates/shared_services/firewall/terraform/firewall.tf

```diff
@@ -11,11 +11,11 @@ resource "azurerm_public_ip" "fwpip" {
 resource "azurerm_firewall" "fw" {
   depends_on          = [azurerm_public_ip.fwpip]
   name                = "fw-${var.tre_id}"
-  resource_group_name = var.resource_group_name
+  resource_group_name = local.core_resource_group_name
   location            = var.location
   ip_configuration {
     name                 = "fw-ip-configuration"
-    subnet_id            = var.firewall_subnet.id
+    subnet_id            = data.azurerm_subnet.firewall.id
     public_ip_address_id = azurerm_public_ip.fwpip.id
   }
 
```

Comment:
Discussion point: 
Do we still need lock on the Firewall (below)?  I wasn't sure if it was added as a workaround to the problem of the Firewall being recreated (see https://github.com/microsoft/AzureTRE/issues/1165) or for another reason

Reply:
I think a do not delete is still a good thing. Given it is stateful we need to prevent accidental deletion, but not update.

Reply:
@marrobi what state is saved in the firewall?
The aggregate of all the interactions with it?